src/docker.test.ts

```diff
@@ -230,24 +247,29 @@ describe("Docker images", (): void => {
   );
 
   testProp(
-    "are saved unless cache hit, in read-only mode, or new Docker image list is empty",
+    "are saved unless cache hit, in read-only mode, a previous save occurs on " +
```

Comment:
I would summarize the third case as "cache already saved" and keep the comma before "or" for consistency.